code/modules/modular_computers/computers/item/computer.dm

```diff
@@ -148,7 +148,12 @@
 
 /obj/item/modular_computer/microwave_act(obj/machinery/microwave/M)
 	. = ..()
-	src.give_power(999999) //The power proc itself will cap this at the cell's max charge
+	var/obj/item/computer_hardware/battery/battery_module = all_components[MC_CELL]
+	if(battery_module && battery_module.battery && battery_module.battery.charge < battery_module.battery.maxcharge * 0.4)
+		src.give_power(battery_module.battery.maxcharge * 0.4 - battery_module.battery.charge)
+	if(prob(5))
+		explosion(src, 0, 1, 4, 5)
+		message_admins("A microwave was detonated (5% chance) at [ADMIN_JMP(src)]")
```

Comment:
add a log_game here. Ideally print the name of the computer that was inserted. Also explain why it was detonated (due to a modular computer being inside)